static/app/components/core/textarea/index.tsx

```diff
@@ -53,3 +53,21 @@ export const TextArea = styled(Input.withComponent(TextAreaControl), {
       min-height: unset;
     `}
 ` as unknown as typeof TextAreaControl;
+
+export const TextArea = withChonk(
+  StyledTextArea,
+  /* re-set height to let it be determined by the rows prop */
+  chonkStyled(StyledTextArea)`
+    height: unset;
+    ${({theme, size = 'md'}) => `padding-top: calc(
+      (${theme.form[size].height} -
+        (${theme.form[size].fontSize} * ${theme.text.lineHeightBody})
+      ) / 2
+    )`};
+    ${({theme, size = 'md'}) => `padding-bottom: calc(
+      (${theme.form[size].height} -
+        (${theme.form[size].fontSize} * ${theme.text.lineHeightBody})
+      ) / 2
+    )`};
```

Comment:
this calculation reduces padding to account for the line-height, to ensure text inside a textArea is still correctly centered.

since it’s a calc expression, it will work across sizings.

Reply:
@TkDodo ❤️ Imo worth leaving a code comment here as well with the same explanation

Reply:
sure, can I just write comments like `/* ... */` inside the styled components template string? will that be filtered out?

Reply:
will this work?

2a0fe3f0